app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java

```diff
@@ -0,0 +1,30 @@
+package com.example.pps_tudai.mvp.view;
+
+import android.app.Activity;
+import android.widget.Toast;
+import java.lang.ref.WeakReference;
+import butterknife.ButterKnife;
+
+public class MainView {
+
+    // activity should never be exposed publicly
+    private WeakReference<Activity> activityWeak;
+
+    public MainView(Activity activity) {
+        ButterKnife.bind(this, activity);
+        this.activityWeak = new WeakReference<Activity>(activity);
+    }
+
+    public void showWelcomeMessage() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Welcome to the app", Toast.LENGTH_LONG).show();
```

Comment:
no hardcode pls.. here you can use a string resource and access it by R.string.YOUR_STRING_NAME

Reply:
done!